src/libs/cozyAppBundle/cozyAppBundle.ts

```diff
@@ -0,0 +1,268 @@
+    folder: stackVersion + tarPrefix,
+    fqdn,
+    slug,
+    version: stackVersion
+  })
+
+  await handleCleanup({
+    client,
+    slug,
+    versionsToKeep: getVersionsToKeep({ type, currentVersion, stackVersion })
+  })
+}
+
+const deleteVersionBundleFromLocalFilesIfExists = async ({
+  client,
+  slug,
+  tarPrefix,
+  version
+}: AppInfo & { tarPrefix: string; version: string }): Promise<void> => {
+  log.debug(`Check if local '${slug}' bundle version exist for '${version}'`)
+
+  const expectedVersionPath = getCozyAppFolderPathForVersion({
+    client,
+    slug,
+    version
+  })
+  const expectedVersionPathWithTarPrefix = `${expectedVersionPath}${tarPrefix}`
+
+  const doesFolderExists = await RNFS.exists(expectedVersionPathWithTarPrefix)
+  if (doesFolderExists) {
+    log.debug(
+      `Local '${slug}' bundle for version '${version}' already existing, deleting existing folder before download`
+    )
+    await RNFS.unlink(expectedVersionPathWithTarPrefix)
+  }
+}
+
+const getCozyAppFolderPathForVersion = ({
+  client,
+  slug,
+  version
+}: AppInfo & { version: string }): string => {
+  const { fqdn } = getFqdnFromClient(client)
+
+  const baseFolderForFqdnAndSlug = getBaseFolderForFqdnAndSlug(fqdn, slug)
+
+  return `${baseFolderForFqdnAndSlug}/${version}`
+}
+
+const getCozyAppArchivePathForVersion = ({
+  client,
+  slug,
+  version
+}: AppInfo & { version: string }): string => {
+  const { fqdn } = getFqdnFromClient(client)
+
+  const baseFolderForFqdnAndSlug = getBaseFolderForFqdnAndSlug(fqdn, slug)
+
+  return `${baseFolderForFqdnAndSlug}/${version}.tar.gz`
+}
+
+const downloadAndExtractCozyAppVersion = async ({
+  client,
+  destinationPath,
+  slug,
+  type,
+  version
+}: AppInfo & {
+  destinationPath: string
+  type: AppType
+  version: string
+}): Promise<void> => {
+  log.debug(`Downloading '${slug}' version '${version}' from stack`)
+
+  const archivePath = getCozyAppArchivePathForVersion({
+    client,
+    slug,
+    version
+  })
+  await RNFS.mkdir(destinationPath)
+  await downloadCozyAppArchive({
+    client,
+    destinationPath: archivePath,
+    slug,
+    type,
+    version
+  })
+
+  await extractCozyAppArchive(archivePath, destinationPath)
+
+  await removeCozyAppArchive(archivePath)
+}
+
+const extractCozyAppArchive = async (
+  archivePath: string,
+  destinationPath: string
+): Promise<void> => {
+  try {
+    await RNFS.mkdir(destinationPath)
+    await Gzip.unGzipTar(archivePath, destinationPath, true)
+  } catch (err) {
+    log.error(`Error while extracting archive: ${getErrorMessage(err)}`)
+    throw err
+  }
+}
+
+const removeCozyAppArchive = async (archivePath: string): Promise<void> => {
+  await RNFS.unlink(archivePath)
+}
+
+const downloadCozyAppArchive = async ({
+  client,
+  destinationPath,
+  slug,
+  type,
+  version
+}: AppInfo & {
+  destinationPath: string
+  type: AppType
+  version: string
+}): Promise<void> => {
+  const stackClient = client.getStackClient() as {
```

Comment:
Can we use inline type here, as in https://github.com/cozy/cozy-flagship-app/blob/master/src/libs/intents/localMethods.ts#L59 ?

Reply:
Improved cozy client typings app-wide